README.md

```diff
@@ -37,6 +37,142 @@ Third-party apps should link to the same single library `implementation(project(
 
 There is no AIDL layer for push notifications. Instead of an AIDL-based system service, OEMs implement their own [distributor](https://unifiedpush.org/users/distributors/) which is reacting to a specific broadcast.
 
+In order to send/receive push notifications a distributor is needed. In this case, we are using Sunup in order to be able to test. Sunup is already doing some work for us, since Mozzilla push services are set by default. We also need some kind of service that handles the Web Pushes: send, receive, register a push service and unregister push service.
+
+Add the following dependency to your `build.gradle`:
+```
+(TBD)
+```
+
+Create a class that implements `PushService` from our SDK:
+```kotlin
+class PushServiceImpl: PushService() {
+    class NewRegistrationState(val registered: Boolean)
+
+    /**
+     * A new endpoint is to be used for sending push messages. The new endpoint
+     * should be send to the application server, and the app should sync for
+     * missing notifications.
+     */
+    override suspend fun onNewEndpoint(endpoint: PushEndpoint, instance: String) {
+        // TODO: Send the new endpoint to your web push server
+        updateRegistrationState(true)
+    }
+
+   /**
+     * A new message is received. The message contains the decrypted content of the push message
+     * for the instance
+     */
+    override suspend fun onMessage(message: PushMessage, instance: String) {
+        // TODO: Create and show a notification with the message received
+        updateRegistrationState(true)
+    }
+
+    /**
+     * The registration is not possible, eg. no network, depending on the reason,
+     * you can try to register again directly.
+     */
+    override suspend fun onRegistrationFailed(reason: FailedReason, instance: String) {
+      // TODO: Inform the user that the registration failed and therefore notifications
+      // are not available
+    }
+
+    /**
+     * This application is unregistered by the distributor from receiving push messages
+     */
+    override suspend fun onUnregistered(instance: String) {
+        // TODO: Send an unregister action to your web push server, removing the endpoint
+        updateRegistrationState(false)
+    }
+
+    /**
+     * Update the UI
+     */
+    private suspend fun updateRegistrationState(registered: Boolean) {
+        _events.emit(NewRegistrationState(registered))
+    }
+
+    companion object {
+        private const val TAG = "PushServiceImpl"
+        private val _events = MutableSharedFlow<NewRegistrationState>()
+        val events = _events.asSharedFlow()
+    }
+}
+```
+
+Add your `PushService` implementation to your manifest:
+```xml
+<service android:name="global.covesa.sdk.client.push.PushServiceImpl"
+    android:exported="true">
+    <intent-filter>
+        <action android:name="global.covesa.sdk.PUSH_EVENT"/>
+        <action android:name="org.unifiedpush.android.connector.RAISE_TO_FOREGROUND"/>
+    </intent-filter>
+</service>
```

Comment:
```suggestion
<service android:name="global.covesa.sdk.client.push.PushServiceImpl"
    android:exported="false">
    <intent-filter>
        <action android:name="global.covesa.sdk.PUSH_EVENT"/>
    </intent-filter>
</service>
```

This should not be exported, and the (dummy) service with action RAISE_TO_FOREGROUND is exposed by the lib